src/main/app/servers.ts

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {IpcMainEvent, ipcMain} from 'electron';
+
+import {MattermostTeam, Team} from 'types/config';
+
+import {UPDATE_SHORTCUT_MENU} from 'common/communication';
+import {Logger} from 'common/log';
+import ServerManager from 'common/servers/serverManager';
+
+import ViewManager from 'main/views/viewManager';
+import ModalManager from 'main/views/modalManager';
+import MainWindow from 'main/windows/mainWindow';
+import {getLocalPreload, getLocalURLString} from 'main/utils';
+
+const log = new Logger('App.Servers');
+
+export const switchServer = (serverId: string, waitForViewToExist = false) => {
+    ServerManager.getServerLog(serverId, 'WindowManager').debug('switchServer');
+    MainWindow.show();
+    const server = ServerManager.getServer(serverId);
+    if (!server) {
+        ServerManager.getServerLog(serverId, 'WindowManager').error('Cannot find server in config');
+        return;
+    }
+    const nextTab = ServerManager.getLastActiveTabForServer(serverId);
+    if (waitForViewToExist) {
+        const timeout = setInterval(() => {
+            if (ViewManager.getView(nextTab.id)) {
+                ViewManager.showById(nextTab.id);
+                clearTimeout(timeout);
+            }
+        }, 100);
```

Comment:
can you explain this value of 100ms?

Reply:
There was an issue where sometimes the view wouldn't exist after a new server was added, due to a race condition. We picked an arbitrary value.

I'm not sure if the problem still exists but I wanted to keep the logic the way it is during the refactor.

Reply:
I'm ok with the logic, just want to make sure this value is not an overkill. 0/5 if we change it now or later.